### DISCUSSION TITLE
What are your top four tools?

### DISCUSSION THREAD

**Discussion by fearlessrob** [FearlessSolutions/engineering-practice-domains#18] [Ideas]
What are the top four tools you use for staying productive during the day? It can be a method or system, application, or just a state of mind. You can include development tools if they really help, but avoid IDEs for the purpose of this discussion. Link out to a website or describe them in a sentence but keep your responses brief so that folks reading this thread can scroll through quickly and get some inspiration.

**Comment by fearlessrob** [comment]
My top four tools for staying focused and productive during the workday are:

* [Toggl](https://toggl.com/) (free version) for passive time tracking in my browser so updating Unanet at COB is a breeze
* The [pomodoro method](https://todoist.com/productivity-methods/pomodoro-technique) to stay focused during coding sessions in 25 minute intervals and force myself to take breaks and stand up
* [Joplin app](https://joplinapp.org/) for contemporaneous note taking in Markdown that "just works"
* [Todoist](https://todoist.com/) to rapidly sort, assess, and process miscellaneous tasks using this [flow diagram](https://www.moehrbetter.com/gtd-advanced-workflow-diagram.html)

**Reply by thwalker6** [reply]
Man you love markdown

**Comment by thwalker6** [comment]
I'm really system and routine based.  I am not the type of person who can use  a tool to track my thoughts and actions but I can get myself going with certain things.

Music -  I need to get in a flow, it works.  I have built many of applications listening to Queen
Exercise - I will do stretches, lift some weights, kick a target
Annoying my wife - I get stuck on an issue or I need to complain I will talk to my wife about it.  She has literally zero idea what I'm talking about, hates when I do this, but it gets my gears turning.
Aderall - I have ADHD and if I have to do something I really dont want to do, documentation for example, it gets me what I need to do it.  Give me a complex problem that can take 1 week to solve I wont move, ask me to update a Jira ticket with A.C. might as well pull my teeth.

**Reply by thetif** [reply]
I was definitely going to say Metadate, but I wasn't sure anyone would be familiar with the name 😄

**Reply by thetif** [reply]
also +1000000 to this
> Give me a complex problem that can take 1 week to solve I wont move, ask me to update a Jira ticket with A.C. might as well pull my teeth.

**Reply by fearlessrob** [reply]
That's what the GTD method helps with. Process your "inbox" (physical or digital) from the top down and you're not allowed to skip. Adding an AC takes about 2 minutes. That means if  you see it you just do it right away so you can move to the next thing. It's helped me a lot.

**Reply by mikewolfd** [reply]
I've started doing this, for me it was acknowledging the anxiety present about starting something and that it was irrational and I just need to move through the things in order

**Comment by thetif** [comment]
- [Todoist](https://todoist.com/) for all the things I need to do
- [Obsidian](https://obsidian.md/) for note taking/info dumping/thinking outloud (I have templates if anyone wants them)
- [InYourFace](https://inyourface.app/) for breaking out of hyperfocus to actually attend meetings
- [Timery](https://timeryapp.com/) for making [Toggl](https://toggl.com/) slightly easier to use to track time

**Reply by emanguy** [reply]
Obisidian is a great one. Once you get a system going it's such a fantastic knowledge base for yourself

**Reply by emanguy** [reply]
Todoist sections and boards have been awesome in a couple cases too.

I've been using todoist boards to keep track of merge requests I need to be aware of:
![image](https://github.com/user-attachments/assets/34032db4-76ac-43e1-9187-694defc30915)

And then I use sections to keep track of information I need to bring up at regular cross-team meetings:
![image](https://github.com/user-attachments/assets/db7e7aa9-970b-442f-b523-7b9b39af397e)

**Comment by mikewolfd** [comment]
I feel very basic here
- Apple Notes + Google Keep 
- ChatGPT
- Reclaim for cross-calendar syncing
- ????
- Profit!!!

But seriously,  I always try to use cool tools, but I just end up doing too much there, and I feel overwhelmed or don't want to deal with configuring it or supporting it, and just go back to the simplest shit.

**Reply by thetif** [reply]
oh I forgot about Reclaim, that's kind of so clutch that I don't even have to think about it

**Reply by PhilR8** [reply]
> But seriously, I always try to use cool tools, but I just end up doing too much there, and I feel overwhelmed or don't want to deal with configuring it or supporting it, and just go back to the simplest shit.

This is me.  I've gone back to a pencil and notebook for note taking because it's as simple as it gets.  It lets me chunk out stories and features into lists, it lets me doodle to quiet the restless side of my mind when I'm in meetings, it's easy to go back and browse through notes from previous days.  

Sometimes I've thought about getting a Remarkable tablet for the best of both worlds -- the tactile feel of writing by hand while also having searchable notes -- but I feel like I'm not at a point in my career/job responsibilities where it's necessary.

**Reply by fearlessrob** [reply]
@PhilR8 That's my jam too. I look at Obsidian and I think "This could do a lot for my notetaking, organizing, mindmapping, etc etc etc" and then I remember that note taking is literally all Joplin does and that's the tool I want at hand when things get stressful.

**Reply by thetif** [reply]
I got all hyperfocused on Obsidian and I looooooooove the dataview plugin because it let's me query all my notes and make tables and lists and stuff out of that data

![CleanShot 2024-09-11 at 09 23 12](https://github.com/user-attachments/assets/30cd7096-648e-41bc-a14f-190482c366fd)

**Reply by emanguy** [reply]
The dataview plugin has also been really good for me in terms of tracking meetings I've had and the notes on those meetings:

![image](https://github.com/user-attachments/assets/7588edf5-db1f-474a-ab19-051f6f6a9496)

**Comment by ZacharyRizer** [comment]
I feel very basic as well:

- Neovim + TMUX - this is the biggest factor of my productivity and workflow
- ChatGPT - becoming more important for how I dig my way through issues
- And Gmail/GoogleCalendar

That's 90% of my workday

**Reply by fearlessrob** [reply]
@ZacharyRizer Can you hook me up with some tutorials on Neovim/TMUX basics?

**Reply by ZacharyRizer** [reply]
@fearlessrob so as you'd expect, theres a billion ways to configure Neovim. That's part of what makes it great, and also what makes it annoying. Also the hot new plugins change on the daily. However you don't need anything but the basic plugins to make it great. The most important part is Vim motions, but I think you already are familiar with Vim motions. Once you have those down, you just need to configure what you want/need in Neovim. 

The basic setup (beyond raw Neovim) in my opinion would be LSP (the most complex aspect), a Completion engine (and snippets if you use them), and VERY important IMHO is a solid Fuzzy Finder (Telescope has pretty much become the go to fuzzy finder for Neovim) . I also have plugins for linting, formatting, git integration, some UI aspects (indent guides, colorscheme, a file tree for viewing large project structure, and a few others. 

This video is a solid start for setup: 

https://www.youtube.com/watch?v=6pAG3BHurdM

And here is a good TMUX integration: 

https://www.youtube.com/watch?v=_YaI2vDbk0o&t=100s

Tmux is its own beast in itself, but I think it is what truly makes Neovim so tempting. You can go a LONG way with just vim motions in your editor of choice like VsCode or Intellij. But the all in one terminal integration that TMUX brings along with syncing motions across all TMUX panes, windows, and Vim buffers is amazing. I literally only use my Terminal and my Browser for 90% of what I do through the day. 

You can also take a look at my config for a starting place. I have it setup very modular, so you can just remove the plugin file for the plugins. you don't want or need. 

https://github.com/ZacharyRizer/.config/tree/main/nvim

**Reply by thetif** [reply]
coming over here to ask the same question

**Reply by ZacharyRizer** [reply]
@thetif I am always happy to nerd out about Neovim, lol. Feel free to message with any question.

**Comment by tbolt** [comment]
MacOS - Since '99
Slack - Not ideal but the alternatives are 🤢 
Notes.app - Moved from Evernote to it ~10 years ago and it's been great
1Password

Notable mentions: 
Reminders.app because I need to be reminded often 😄 
GitHub

**Comment by jphechter** [comment]
**Sublime/MacOS Stickies** - I'm an awful note taker, and I got in a really bad habit of not taking notes during meetings because I'd open a Google Doc, jot a few incoherent sentences, and then trash the doc because it didn't really serve any long term purpose. The incoherent sentences did mean something to me though. I need a way to keep half-baked ideas around until they can be turned into a ticket, a meeting, something more substantial with greater shelf-life. Sublime is awesome and I treat it like a legal pad. The MacOS Stickies app is just as powerful, and has the added advantage of being able to "float" a note on-top of every other application window. Even if I move to another task, the note is top level on my screen.

**Multiple Desktops** - I have 3 monitors, and they help, but I would be lost without the ability to have multiple desktops. This has been extremely helpful for organizing my work. I typically try to categorize things on each desktop so I can switch screens and tasks.

**ChatGPT** - I honestly need to use it more. Great as a rubber duck, code generation tool, and code analysis tool.

**Calendar** - Don't really care which calendar app, but I live and die by the schedule.
- Time keeping - I create a separate calendar from the one I use for meetings so I can track time. I already have a list of all the meetings I will be/was in. This is to actually track tasks that were accomplished throughout the day. This is really helpful for planning the future, but also reporting back in standups. Even if I don't have explicit notes, just knowing what happened during a day is invaluable.
- Planning - I was recently introduced to some ideas out of a book called [The Staff Engineer's Path](https://www.oreilly.com/library/view/the-staff-engineers/9781098118723/) (courtesy of @alexander-littleton). One of the ideas suggested is baking the tasks you plan to do into your schedule. Setting an intention and acknowledging how finite time is as a resource has helped me prioritize my work.

![Screenshot 2024-09-04 at 5 06 09 PM](https://github.com/user-attachments/assets/d145743b-daf1-47b3-8948-6958829c604b)
![Screenshot 2024-09-04 at 5 06 18 PM](https://github.com/user-attachments/assets/a2b057ff-87b7-4fa5-880e-3a2af631721d)

**Reply by thetif** [reply]
thanks for the req, just got the audiobook, it looks cool

**Reply by lexlittleton** [reply]
It's accidentally the best book on engineering soft skills I think I've read. The content seems just as valuable for a Junior as it is for a Principal.

**Comment by peggles2** [comment]
Ok I also am a bit basic here are my tools but some I need to find a replacement for:
- Evernote ( now it is no longer free )
- vim
- RubyMine, Pycharm, visual studio code
- ChatGPT
- slack
- google calendar

**Reply by fearlessrob** [reply]
@peggles2 Nothing wrong with basic.

**Comment by cyber-decker** [comment]
I think there's very few tools I come back to over and over again and use on a daily basis.  Maybe similar to how I like to work in different environments as frequently as I can (in offfice, out of office, at home, from the desk, from the couch, sitting on the ground) I sort of like to be in different working tools often as well.  I find this helps me to approach problems in different ways with different mindsets.

Maybe that speaks a lot to the fact that I don't have a consistent way of delivering results, which could be a good or bad thing, but at least I get a variety of results and ways of doing things.

That being said, there is one tool I come back to over and over.

[XMind](https://xmind.app/) - this is probably my favorite note-taking app for mind mapping.  I find the interface to be very easy for me to use, simple to navigate, easy to learn and distraction free. I've been using mind mapping tools for about 10 years or so, and this is my favorite for everything from taking notes during a meeting to ideation on testing to exploring thoughts and concepts very fluidly just to get it out there. I don't often keep a lot of what I just want to get out but it's certainly a great outlet for thoughts.

+1's to other apps already mentioned:
- 1Password - indispensable tool for password management for myself and my family.  Don't know what I would do without this.
- Obsidian - great markdown editing.  Love this editor so much for markdown.
- Google Keep - handy for quick notes, shared notes, grocery lists, etc.
- Google Calendar - couldn't get anything done without this.  Having shared calendars for the family is amazing.
